sklearn/utils/_array_api.py

```diff
@@ -500,9 +561,20 @@ def move_to(*arrays, xp, device):
             "namespace is Numpy"
         )
 
+    arrays_ = arrays
+    # Down cast float64 `arrays` when highest precision of `xp`/`device` is float32
+    if _max_precision_float_dtype(xp, device) == xp.float32:
+        arrays_ = []
+        for array in arrays:
+            xp_array, _ = get_namespace(array)
+            if getattr(array, "dtype", None) == xp_array.float64:
+                arrays_.append(xp_array.astype(array, xp_array.float32))
+            else:
+                arrays_.append(array)
```

Comment:
The previous comment about this suggestion was marked as outdated because of a code change, but the discussion is still valid:

https://github.com/scikit-learn/scikit-learn/pull/32755#discussion_r2611300838